lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1097,7 +1097,14 @@ def __init__(
         self._relation_name = relation_name
         self._tool = CosTool(self)
         self.port = int(port)
-        self.container = self._charm._container
+
+        # This needs to return none if it does not exist for
+        # compatibility with machine charms. Also, this should
+        # be removed in v1 of the library as we should not
+        # assume any charm attributes exist except those
+        # guaranteed by ops.
+        self.container = getattr(self._charm, "_container", None)
```

Comment:
More likely than not  this is dead code and can probably be removed.
`container` comes from the charm. Why would any charm code access its container via the lib?
I vote for removal.

Reply:
I agree but I was hesitant due to the lack of an underscore. I think you are right though. I doubt anything is using this.